README.rst

```diff
@@ -63,13 +65,27 @@ Settings list
 -------------
 All settings are contained in a ``PUSH_NOTIFICATIONS_SETTINGS`` dict.
 
-In order to use GCM, you are required to include ``GCM_API_KEY``.
-For APNS, you are required to include ``APNS_CERTIFICATE``.
+In order to use GCM, you are required to include one of ``GCM_API_KEY``, ``GCM_API_KEYS``, or ``GCM_API_KEYS_MODEL``.
+For APNS, you are required to include ``APNS_CERTIFICATE``, ``APNS_CERTIFICATES``, or ``APNS_CERTIFICATES_MODEL``.
 
 - ``APNS_CERTIFICATE``: Absolute path to your APNS certificate file. Certificates with passphrases are not supported.
+- ``APNS_CERTIFICATES``: A dictionary reflecting separate application IDs to separate APNS certificate files.
+- ``APNS_CERTIFICATES_MODEL``: A dictionary containing description of a database model to reflect
```

Comment:
Sorry for the delayed response on this.

I can see how your situation is somewhat restrictive! I'll admit that perhaps I don't fully understand your project's workflow, however I still don't see the necessity to have to customize the `push_notifications` model.

I *assume* (and could very well be wrong) that you've got a `Partner` model that is specific to your business logic. Your `Partner` model must be associated with the 3rd party OAuth2 package's `OAuth2Application` model somehow, via a `ForeignKey` maybe. 

Why couldn't your `Partner` model also be associated with the `push_notifications` `Application` model via a `OneToOneField` (or `ForeignKey`) the same way? This way your `Partner` model can easily be used to associate `push_notifications` data to the OAuth2 data.

Based on my **assumptions(!!!!!)**, this use case feels very much like an edge case. However, if still deemed necessary to include in the project, I think that the following suggestion from earlier in the conversation makes sense:

```
# default model setting
PUSH_NOTIFICATIONS_APPLICATION_MODEL = 'push_notifications.Application'

# developers could choose to derive their own model from this to add 
# additional non-push_notification fields
class AbstractBaseApplication(models.Model):
    # all the apns, fcm/gcm, wns etc fields (and add future platform 
    #fields here)
    ...
    
    class Meta:
        abstract = True

# this would be the default model used, unless 
# PUSH_NOTIFICATIONS_APPLICATION_MODEL was specified in Django 
# project settings
class Application(AbstractBaseApplication):
    pass
```

That way, adding future push platforms is automatically provided for users of `push_notifications` via Django migrations, and **edge cases like yours can still benefit by creating a new migration of your own for your custom `Application` model** if desired.